appscale/agents/base_agent.py

```diff
@@ -261,6 +263,18 @@ def diff(self, list1, list2):
         diffed_list.append(item)
     return diffed_list
 
+  @staticmethod
+  def _gen_machine_id(length=8):
+    """
+    Generates an ID that's meant to be unique for a machine within a
+    deployment.
+
+    Returns:
+      A string specifying a machine ID.
+    """
+    return ''.join(random.choice(string.lowercase + string.digits)
```

Comment:
There is a pyhaikunator available which would warn again the cold land of machine names ....

Reply:
The library we were using worked just fine, but it's not efficient in terms of the number of combinations possible:string length. In order to address the issue this PR is trying to fix, the hostname needs to be less than 7 characters.

Reply:
sigh ... haikunator will be missed